Zend/zend_builtin_functions.c

```diff
@@ -148,6 +148,12 @@ ZEND_FUNCTION(gc_status)
 	add_assoc_long_ex(return_value, "threshold", sizeof("threshold")-1, (long)status.threshold);
 	add_assoc_long_ex(return_value, "buffer_size", sizeof("buffer_size")-1, (long)status.buf_size);
 	add_assoc_long_ex(return_value, "roots", sizeof("roots")-1, (long)status.num_roots);
+
+	/* Using double because zend_long may be too small on some platforms */
+	add_assoc_double_ex(return_value, "application_time", sizeof("application_time")-1, (double) status.application_time / ZEND_NANO_IN_SEC);
```

Comment:
the init size above should be updated

also are/can these strings be interned for better perf?

what is the time measurement and `gc_status()` itself overhead?

Reply:
I've updated the initial array size, thank you.

Regarding the time measurement overhead:

The synthetic benchmarks [here](https://gist.github.com/arnaud-lb/fd550e6b4a9aca5142f1b3299649b35a) and [here](https://gist.github.com/arnaud-lb/4fb3884cf5f14823b27a90a44371973f) show no significant overhead over multiple runs. The number of 10k nodes was chosen because this is the default GC threshold. Applications will typically have more nodes to scan, and their `measurement_overhead/actual_work` ratio will be lower.

I've seen no significant overhead on a long running application.

Regarding the overhead on `gc_status()` itself, it would matter if the function was called a high number of times during a request. Do you think it will be the case?

Reply:
> Regarding the overhead on gc_status() itself, it would matter if the function was called a high number of times during a request. Do you think it will be the case?

In PHPUnit 10, `gc_status()` is called for each event that is emitted. This is "a lot", but IMO perfectly fine in a development context.

Reply:
I see no slowdown on the PHPUnit 10 test suite. However, `gc_status()` was called only 128 times. Is there a configuration that would cause more events to be triggered?

Reply:
I was looking at the wrong file: `gc_status()` is actually called more than 19k times during the test suite and the changes in this PR caused no slowdown.

Reply:
PHPUnit's test suite (`main` branch) currently has 2845 tests. For each test, multiple events are emitted. For each event, `gc_status()` is called. I am confused that you only see 128 calls.

Reply:
Aha! :)